wdl/tasks/QC/AlignedMetrics.wdl

```diff
@@ -528,17 +200,36 @@ task ComputeBedCoverage {
 task ReadMetrics {
     input {
         File bam
+        String? gcs_output_dir
 
         RuntimeAttr? runtime_attr_override
     }
 
+    String? output_dir = if defined(gcs_output_dir) then sub(select_first([gcs_output_dir]), "/?$", "/yield_aligned/") else gcs_output_dir
     String basename = basename(bam, ".bam")
     Int disk_size = 2*ceil(size(bam, "GB"))
 
     command <<<
         set -euxo pipefail
 
         java -jar /usr/local/bin/gatk.jar ComputeLongReadMetrics -I ~{bam} -O ~{basename}.read_metrics -DF WellformedReadFilter
+        samtools flagstat ~{bam} > ~{basename}.flag_stats.txt
```

Comment:
This is great.
Should it have its own task, though? I know it's always a balance (localization vs. separation of concerns).

Reply:
There's an astounding amount of overhead in invoking a task.  Provisioning the VM, downloading the docker image, checking the call cache database, localizing input files, etc., etc., etc.  It's especially true of small tasks.  For example, in this workflow the task MakeChrIntervalList (an extreme example, I admit) runs for 2 minutes.  Of this, the useful work part (UserAction) takes 1.36 seconds.  I'm voting for a rule that says, "You combine all the operations that process the same data into a single task unless it's a very long-running task that might escape preemption or significantly reduce workflow real clock time by being parallelized."